File: Programming EF Code First/chapter5/Chapter5 CSharp/BAGA_FluentAPI/DataAccess/TripConfiguration.cs
using System.Data.Entity.ModelConfiguration;
using Model;
using System.ComponentModel.DataAnnotations;

namespace DataAccess
{
  public class TripConfiguration : EntityTypeConfiguration<Trip>
  {
    public TripConfiguration()
    {
      HasKey(t => t.Identifier);

      Property(t => t.Identifier)
        .HasDatabaseGeneratedOption(DatabaseGeneratedOption.Identity);

      Property(t => t.RowVersion)
        .IsRowVersion();

      HasMany(t => t.Activities)
        .WithMany(a => a.Trips)
        .Map(c =>
        {
          c.ToTable("TripActivities");
          c.MapLeftKey("TripIdentifier");
          c.MapRightKey("ActivityId");
        });
    }
  }
}

File: Programming EF Code First/chapter5/Chapter 5 VB/Fluent/DataAccess/TripConfiguration.vb
Imports System.Data.Entity.ModelConfiguration
Imports Model
Imports System.ComponentModel.DataAnnotations

Namespace DataAccess
  Public Class TripConfiguration
	  Inherits EntityTypeConfiguration(Of Trip)
	Public Sub New()
	  HasKey(Function(t) t.Identifier)

      Me.Property(Function(t) t.Identifier).HasDatabaseGeneratedOption(DatabaseGeneratedOption.Identity)

      Me.Property(Function(t) t.RowVersion).IsRowVersion()

      Me.HasMany(Function(t) t.Activities).WithMany(Function(a) a.Trips).Map(Sub(c)
                                                                               c.ToTable("TripActivities")
                                                                               c.MapLeftKey("TripIdentifier")
                                                                               c.MapRightKey("ActivityId")
                                                                             End Sub)
	End Sub
  End Class
End Namespace